backend/routes/planForm.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import express from 'express'
+import prisma from '../utils/prismaClient'
+import { type User, InviteStatus, Role, EventType } from '@prisma/client'
+import { body } from 'express-validator'
+import { handleInputErrors } from '../modules/middleware'
+const planFormRouter = express.Router()
+const db = prisma
+
+planFormRouter.post('/',
+  body('description').isString(),
+  body('startDate').isString(),
+  body('startDate').isString(),
+  body('endDate').isString(),
+  body('EventType').isIn(Object.values(EventType)),
+  body('friends').isArray(),
+  handleInputErrors,
+  async (req, res) => {
+    const {
+      description,
+      startDate,
+      endDate,
+      EventType,
+      friends
+    } = req.body
+    try {
+      const plan = await db.plan.create({
```

Comment:
so in general we don't usually want a route file like this to contain the code that directly modifies the db. a standard name for this is called a "3 layer service architecture" 

see this article: https://blog.devgenius.io/the-three-layer-architecture-for-node-js-applications-ce32a3a30fa6

the names are relatively interchangeable but typically I see/use "routes" "controllers" and "models" as the main 3 layers. 

In this route file, we receive and validate the request schema and in the request handler we are passed the `req` and `res`. generally we never pass the `req` and `res` further down and leave it in this file. 

we pull out the values we need, and pass them into `controller` methods. in this example i'd expect there to be a `backend/controllers/plan.ts` file of some sort and it would have a method called like `createPlan` that takes in the values in your `req.body`.  generally all of the true business logic lives inside the controller file

And then finally we have `model` files, which would house all the code that directly calls the prisma client and calls the actual db methods. all files wishing to make db updates would then be able to go through methods in this file, rather than having db calls just spread out everywhere